dbt/adapters/athena/impl.py

```diff
@@ -135,7 +135,25 @@ def _get_one_catalog(
         )
 
         results = self._catalog_filter_table(table, manifest)
-        return results
+
+        table_owners = []
+        # Get the owner for each model from the manifest
+        for node in manifest.nodes.values():
+            if node.resource_type == "model":
+                table_owners.append({
+                    "table_name": node.alias,
+                    "table_owner": node.config.meta.get("owner")
+                })
+        owners = agate.Table.from_object(table_owners)
+
+        # Join owners with the results from catalog
+        results_with_owners = results.join(
+            right_table=owners,
+            left_key="table_name",
+            right_key="table_name",
+        )
```

Comment:
could you put this in separate private functions and try to add unit tests??

Reply:
I can break it down into functions, but I unless I'm being really stupid (not impossible 😅 ) I don't see any existing unit tests for this repo?

Reply:
You are definitely not stupid, unit tests need to be added. Ok, let's break it a bit down in small functions, then we can cover the unit testing later on.

Reply:
factored it out into a private function. I also realised that the join was missing the schema and database as join keys so added those in